tests/acceptance/10_files/copy_from_this_promiser.cf

```diff
@@ -0,0 +1,85 @@
+#!/var/cfengine/bin/cf-agent -f-
+body common control
+{
+        inputs => { "../default.cf.sub" };
+        bundlesequence  => { default("$(this.promise_filename)") };
+        version => "1.0";
+}
+
+
+bundle agent init {
+  vars:
+      "ROOT" string => "/tmp/CFE-2489";
+      "SOURCE" string => "$(ROOT)/SOURCE";
+      "use_bundles"
+        slist => { "bundle_uses_copy_from_without_param",
+                   "bundle_uses_copy_from_with_param" };
+
+  files:
+      "$(init.ROOT)/$(use_bundles)/." create => "true";
+
+      # Setup source files to copy_from
+      "$(init.ROOT)/SOURCE/$(init.ROOT)/$(use_bundles)/file.txt"
+        create => "true";
```

Comment:
this path looks real weird. Shouldn't it just be `"$(SOURCE)/$(use_bundles)/file.txt"`?

Reply:
No, the case is using $(this.promiser) inside the copy_from body.
Basically, the use case is like: Promise file (`/etc/myfile`) is a copy from the same path in a remote location. So under SOURCE, I needed to seed the full path to the file.

That way you could do something like this:
```
/etc/profile
  when in prod, copy from /SOURCE/PROD/etc/profile
  when in dev, copy from /SOURCE/DEV/etc/profile
```
Or perhaps:
```
/path/$(application)/config
  when talking about app1, copy from /SOURCE/app1/config
  when talking about app2, copy from /SOURCE/app2/config.
```

Reply:
Right. I see it now. `$(init.ROOT)/SOURCE` is like  the root of a tarball/package that would be unpacked/copied to `/`. Ack.